src/Core/Form/IdentifiableObject/OptionProvider/CustomerAddressFormOptionsProvider.php

```diff
@@ -43,27 +43,33 @@ class CustomerAddressFormOptionsProvider implements FormOptionsProviderInterface
      */
     private $queryBus;
 
+    /**
+     * @var array
+     */
+    private $requiredFields;
+
     /**
      * @param CommandBusInterface $queryBus
      */
     public function __construct(CommandBusInterface $queryBus)
     {
         $this->queryBus = $queryBus;
+        $this->requiredFields = $this->queryBus->handle(new GetRequiredFieldsForAddress());
```

Comment:
thats not what I meant.
Just create a private method so you can reuse the whole line (including the 'requiredFields' array key). So like this:
```
    private function getRequiredFieldsOption(): array
    {
        return ['requiredFields' => $this->queryBus->handle(new GetRequiredFieldsForAddress())];
    }
```

Reply:
Hi @zuk3975 I understand what you mean but @kpodemski  asks me for this way. 

I don't know which one is better?

https://github.com/PrestaShop/PrestaShop/pull/29114#discussion_r925974517

Thank for your time

Reply:
@kpodemski could you elaborate why you would want to do that? I don't think its the right way :thinking: